data-prepper-plugins/date-processor/src/test/java/com/amazon/dataprepper/plugins/processor/date/DateProcessorTests.java

```diff
@@ -0,0 +1,369 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.processor.date;
+
+import com.amazon.dataprepper.metrics.PluginMetrics;
+import com.amazon.dataprepper.model.event.Event;
+import com.amazon.dataprepper.model.event.JacksonEvent;
+import com.amazon.dataprepper.model.record.Record;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.hamcrest.Matchers.lessThanOrEqualTo;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.when;
+
+
+@ExtendWith(MockitoExtension.class)
+class DateProcessorTests {
+    @Mock
+    private PluginMetrics pluginMetrics;
+
+    @Mock
+    private DateProcessorConfig mockDateProcessorConfig;
+
+    private DateProcessor dateProcessor;
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final TypeReference<Map<String, Object>> MAP_TYPE_REFERENCE = new TypeReference<Map<String, Object>>() {};
+    private static final String TIMESTAMP_KEY = "@timestamp";
+
+    private final String messageInput =  UUID.randomUUID().toString();
+    private Map<String, Object> testData;
+    private Map<String, Object> resultData;
+    private final String pattern1 = "yyyy-MM-dd";
+    private final String pattern2 = "yyyy-MMM-dd HH:mm:ss.SSS";
+    private final String pattern3 = "yyyy-MM-dd HH:mm:ss";
+    private final String logDate1 = LocalDate.now().toString();
+    private final String logDate2 = LocalDateTime.now().format(DateTimeFormatter.ofPattern(pattern2));
+    private final String logDate3 = "2022-01-12 15:20:45";
+
+    @BeforeEach
+    void setup() {
+        final DateProcessorConfig dateProcessorConfig = new DateProcessorConfig();
+        lenient().when(mockDateProcessorConfig.getFromTimeReceived()).thenReturn(dateProcessorConfig.getFromTimeReceived());
```

Comment:
The mocks are redundant if they are backed by an instant of the class and it is very hard to know what is actually being tested by using lenient mocks. I would recommend avoiding lenient mocks as much as possible and enforcing the actual stubbing required for each test.